.github/workflows/ci.yml

```diff
@@ -23,9 +23,11 @@ jobs:
     - uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-  
+
     - name: Install and build
-      run: python -m pip install -r requirements-dev.txt
+      run: |
+        python -m pip install -r requirements-dev.txt
+        python -m pip install .
```

Comment:
You are not currently installing the pinned versions of the main packages here because they are not specified in the `requirements-dev.txt`. You can see that pandas etc is being installed at the `python -m pip install .` step in the CI https://github.com/ImperialCollegeLondon/wsi/actions/runs/6875139142/job/18698156546